FirebaseRemoteConfig/SwiftNew/ConfigExperiment.swift

```diff
@@ -0,0 +1,193 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseABTesting
+import Foundation
+
+// TODO(ncooke3): Once everything is ported, the `@objc` and `public` access
+// can be removed.
+
+/// Handles experiment information update and persistence.
+@objc(RCNConfigExperiment) public final class ConfigExperiment: NSObject {
+  private static let experimentMetadataKeyLastStartTime = "last_experiment_start_time"
+  private static let serviceOrigin = "frc"
+
+  @objc private var experimentPayloads: [Data]
+  @objc private var experimentMetadata: [String: Any]?
```

Comment:
Since all of these properties are inititialized in `init`, can they be non-optional?

Reply:
Done. I left a TODO for `ExperimentController` which I left as an optional because it requires some unit test work.